### PR TITLE
Correcting Logic for archive/unarchive collection process

### PR DESCRIPTION
**Pull request by jchavez137**
- Moved logic for Changing status of a collection
- Correcting logic for Archiving/Unarchiving collection to return to its previous status
- Adding logic to properly update dates when changing status of a collection
- Adding Unit test in CollectionController for status change

### CODE REVIEW

**Comment by JohnKallies** [comment]
I'd like to see some kind of addition to an existing test that covers this case

**Comment by JohnKallies** [comment]
> I'd like to see some kind of addition to an existing test that covers this case

Per our meeting time, I'd like to see a unit test that flexes the execution paths in this case statement,

File: api/src/main/kotlin/edu/wgu/osmt/collection/Collection.kt [hunk 1]
```diff
@@ -76,6 +76,39 @@ data class CollectionUpdateObject(
...
+            PublishStatus.Draft -> {
+                dao.status = PublishStatus.Draft
+                dao.archiveDate = null
+                dao.publishDate = null
+
+            }
+            PublishStatus.Workspace -> {
+                dao.status = PublishStatus.Workspace
+            }
+            else -> {}
```
Comment (JohnKallies) [inline api/src/main/kotlin/edu/wgu/osmt/collection/Collection.kt:108]: Can this case ever happen? CollectionDao.status is not nullable, and the dao argument is also not nullable.

File: api/src/test/kotlin/edu/wgu/osmt/collection/CollectionControllerTest.kt [hunk 2]
```diff
@@ -31,6 +39,8 @@ internal class CollectionControllerTest @Autowired constructor(
     @Autowired
     lateinit var collectionController: CollectionController
 
+    var authentication: Authentication = Mockito.mock(Authentication::class.java)
```
Comment (JohnKallies) [inline api/src/test/kotlin/edu/wgu/osmt/collection/CollectionControllerTest.kt:42]: In Kotlin, we often use mockk. I can't give you a deep reason why, but there are times/places where Mockito is problematic. I'm not asking for a change here, just please be aware.

File: api/src/test/kotlin/edu/wgu/osmt/collection/CollectionControllerTest.kt [hunk 3]
```diff
@@ -124,10 +103,9 @@ internal class CollectionControllerTest @Autowired constructor(
             author = "newAuthor")
         val collection = collectionRepository.create(name = "name", user = "user", email = "j.chavez@wgu.edu", description = "description")
         collection!!.status = PublishStatus.Archived
+        collection!!.archiveDate = LocalDateTime.now()
```
Comment (JohnKallies) [inline api/src/test/kotlin/edu/wgu/osmt/collection/CollectionControllerTest.kt:106]: IDEA reports that collection!! isn't needed on line 106

File: api/src/test/kotlin/edu/wgu/osmt/collection/CollectionControllerTest.kt [hunk 4]
```diff
@@ -70,4 +91,26 @@ internal class CollectionControllerTest @Autowired constructor(
...
+    @Test
+    fun `updateCollection() should return a collection to draft status when providing an Unarchived status and update the archived date`() {
+        // arrange
+        val jwt = Jwt.withTokenValue("foo").header("foo", "foo").claim("email", userEmail).build()
+        val update = ApiCollectionUpdate(
+            name = "newName",
+            description = "newDescription",
+            publishStatus = PublishStatus.Unarchived,
+            author = "newAuthor")
+        val collection = collectionRepository.create(name = "name", user = "user", email = "j.chavez@wgu.edu", description = "description")
```
Comment (JohnKallies) [inline api/src/test/kotlin/edu/wgu/osmt/collection/CollectionControllerTest.kt:104]: Can we please not use real identities in source code?

File: api/src/test/kotlin/edu/wgu/osmt/collection/CollectionControllerTest.kt (hunk 3 above)
Comment (jchavez137) [reply api/src/test/kotlin/edu/wgu/osmt/collection/CollectionControllerTest.kt:106]: Yeah, the assertion is already in line 105, let me remove it.

File: api/src/test/kotlin/edu/wgu/osmt/collection/CollectionUpdateObjectTest.kt [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+): SpringTest(){
+
+
+    @BeforeAll
+    fun setup() {
+
+    }
+
+    @Test
+    fun `applyStatusChange() should apply Published status`() {
```
[the remark this answers is not in the document]
Comment (jchavez137) [reply api/src/test/kotlin/edu/wgu/osmt/collection/CollectionUpdateObjectTest.kt:24]: Correcting

**Pull request merged**